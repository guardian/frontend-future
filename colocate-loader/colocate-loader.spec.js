// @flow
/* eslint-disable global-require, no-console,import/no-dynamic-require */

const path = require('path');
const webpack = require('webpack');
const MemoryFileSystem = require('memory-fs');

const memFs = new MemoryFileSystem();

const distDir = path.resolve(__dirname, 'dist');
memFs.mkdirpSync(distDir);

const compileAndTest = (config, done) => {
	const compiler = webpack(require(`./fixtures/webpack.config.${config}.js`));

	compiler.outputFileSystem = memFs;
	compiler.run(err => {
		if (err) console.log(err);
		const content = memFs.readFileSync(
			path.resolve(distDir, 'main.js'),
			'utf8'
		);
		expect(content).toMatchSnapshot();
		done();
	});
};

it('works without other loaders', done => {
	compileAndTest('no-loaders', done);
});

it('works with other loaders', done => {
	compileAndTest('loaders', done);
});
