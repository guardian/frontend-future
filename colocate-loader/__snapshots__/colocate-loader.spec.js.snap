// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with other loaders 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = 0);
/******/ })
/************************************************************************/
/******/ ([
/* 0 */
/***/ (function(module, exports, __webpack_require__) {


					var style = __webpack_require__(1);
var template = __webpack_require__(2);
var myCrazyBlockName = __webpack_require__(3);
					\\"use strict\\";

var variableToConvertToVar = null;console.log(style);console.log(template);console.log(myCrazyBlockName);
				

/***/ }),
/* 1 */
/***/ (function(module, exports) {

module.exports = {\\".red\\":{\\"color\\":\\"red\\",\\"border-color\\":\\"red\\"}};

/***/ }),
/* 2 */
/***/ (function(module, exports) {

module.exports = \\"<p>hello!</p>\\"

/***/ }),
/* 3 */
/***/ (function(module, exports) {

module.exports = \\"something handy\\"

/***/ })
/******/ ]);"
`;

exports[`works without other loaders 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = 0);
/******/ })
/************************************************************************/
/******/ ([
/* 0 */
/***/ (function(module, exports, __webpack_require__) {


					var style = __webpack_require__(1);
var template = __webpack_require__(2);
var myCrazyBlockName = __webpack_require__(3);
					 const variableToConvertToVar = null; console.log(style); console.log(template); console.log(myCrazyBlockName); 
				

/***/ }),
/* 1 */
/***/ (function(module, exports) {

module.exports = \\" .red { color: red; border-color: red; } \\"

/***/ }),
/* 2 */
/***/ (function(module, exports) {

module.exports = \\"<p>hello!</p>\\"

/***/ }),
/* 3 */
/***/ (function(module, exports) {

module.exports = \\"something handy\\"

/***/ })
/******/ ]);"
`;
